public/components/main/report_details/report_details.tsx

```diff
@@ -75,36 +50,52 @@ export const ReportDetailsComponent = (props) => {
   );
 };
 
-const created_date = new Date('April 20, 2020 20:32:12');
+export function ReportDetails(props) {
+  const [reportDetails, setReportDetails] = useState({});
+  const reportId = props.match['params']['reportId'];
 
-const reportDetailsMockMetadata = {
-  report_name: 'Daily Sales Report-232o2jsf28492h3rjskfbwjk23',
-  description: 'Report Description Here',
-  created: created_date.toString(),
-  last_updated: created_date.toString(),
-  source_type: 'Download',
-  source: 'dashboard/daily_sales',
-  default_file_format: 'PDF',
-  report_header: '--',
-  report_footer: '--',
-  report_type: 'Schedule',
-  schedule_type: 'Now',
-  schedule_details: '--',
-  alert_details: '--',
-  channel: 'Kibana Reports',
-  kibana_recipients: 'admin',
-  email_recipients: '--',
-  email_subject: '--',
-  email_body: '--',
-  report_as_attachment: false,
-};
+  const handleReportDetails = (e) => {
+    setReportDetails(e);
+  };
```

Comment:
if a function only contains another function, can you merge into one?

Reply:
Wrapping useState setter in a handler is standard practice